### PR TITLE
feat: use minio for copy files

### PR DESCRIPTION
**Pull request by vLia**
## Pull request description 

Fixes
https://github.com/kubeshop/testkube/issues/2635
https://github.com/kubeshop/testkube/issues/2501
https://github.com/kubeshop/testkube/issues/2408

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: cmd/kubectl-testkube/commands/tests/create.go [hunk 1]
```diff
@@ -89,6 +89,11 @@ func NewCreateTestsCmd() *cobra.Command {
 
 				err = validateExecutorType(options.Type_, executors)
 				ui.ExitOnError("validating executor type", err)
+
+				if len(copyFiles) > 0 {
+					err := uploadCopyFiles(client, testName, "test", copyFiles)
```
Comment (vsukhin) [inline cmd/kubectl-testkube/commands/tests/create.go:94]: probably, it's enum const

File: cmd/kubectl-testkube/commands/tests/run.go [hunk 2]
```diff
@@ -88,13 +88,16 @@ func NewRunTestCmd() *cobra.Command {
 					os.Exit(1)
 				}
 
+				if len(copyFiles) > 0 {
+					err := uploadCopyFiles(client, name, "execution", copyFiles)
```
Comment (vsukhin) [inline cmd/kubectl-testkube/commands/tests/run.go:92]: probably, worth to have typed const

File: api/v1/testkube.yaml [hunk 3]
```diff
@@ -2100,6 +2100,34 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/Problem"
+  /copy-files:
```
Comment (exu) [inline api/v1/testkube.yaml:2103]: is copy-files a valid resource name?

Comment (exu) [reply api/v1/testkube.yaml:2103]: I would use just `files` and `POST /files`

File: cmd/kubectl-testkube/commands/tests/common.go [hunk 4]
```diff
@@ -386,3 +387,22 @@ func mergeCopyFiles(testFiles []string, executionFiles []string) ([]string, erro
 
 	return result, nil
 }
+
+func uploadCopyFiles(client client.Client, parentID string, parentType string, files []string) error {
```
Comment (exu) [inline cmd/kubectl-testkube/commands/tests/common.go:391]: upload or copy?

File: api/v1/testkube.yaml [hunk 5]
```diff
@@ -2100,6 +2100,34 @@ paths:
...
+      summary: "Upload file"
+      description: "Upload file to be copied into test suite executions"
+      operationId: uploadTestSuiteCopyFile
+      requestBody:
+        $ref: "#/components/requestBodies/CopyFileBody"
+      responses:
+        200:
+          description: "successful operation"
+          content:
+            text/yaml:
```
Comment (vsukhin) [inline api/v1/testkube.yaml:2119]: is this json or yaml response

File: internal/app/api/v1/copyfiles.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		if filePath == "" {
+			return s.Error(c, fiber.StatusBadRequest, errors.New("filePath cannot be empty"))
+		}
+
+		bucketName := getBucketName(parentType, parentID)
+		file, err := c.FormFile("attachment")
+		if err != nil {
+			return s.Error(c, fiber.StatusBadRequest, fmt.Errorf("unable to upload file: %w", err))
+		}
+		f, err := file.Open()
```
Comment (vsukhin) [inline internal/app/api/v1/copyfiles.go:31]: should we have defer f.Close()

**Review by vsukhin — APPROVED** [review]
looks good

File: cmd/kubectl-testkube/commands/tests/create.go (hunk 1 above)
Comment (vLia) [reply cmd/kubectl-testkube/commands/tests/create.go]: updated

File: cmd/kubectl-testkube/commands/tests/run.go (hunk 2 above)
Comment (vLia) [reply cmd/kubectl-testkube/commands/tests/run.go]: updated

File: api/v1/testkube.yaml (hunk 3 above)
Comment (vLia) [reply api/v1/testkube.yaml:2103]: We have agreed on using `--copy-files` in the CLI. `--file` is already used to give the test files as input. (and we also agreed that `--config-file` is too restrictive, as these files will not always be used as configuration files)

While I do agree that copy is a verb in itself, the closest adjective would be `copied`, but that implies it was already copied over. For consistency's sake I'd keep `/copy-files`, unless there is a better and simpler option.

File: api/v1/testkube.yaml (hunk 5 above)
Comment (vLia) [reply api/v1/testkube.yaml:2119]: it is `200`, but I'll add the json parts

File: cmd/kubectl-testkube/commands/tests/common.go (hunk 4 above)
Comment (vLia) [reply cmd/kubectl-testkube/commands/tests/common.go]: updated it to uploadFiles(..)

File: api/v1/testkube.yaml (hunk 3 above)
Comment (vLia) [reply api/v1/testkube.yaml]: After a discussion in a private channel, we decided to go with /uploads for this endpoint. The code is being updated and tested rn

**Pull request merged**